pandas/io/sas/sasreader.py

```diff
@@ -63,14 +71,15 @@ def read_sas(
         else:
             raise ValueError("unable to infer format of SAS file")
 
+    reader: Union["XportReader", "SAS7BDATReader"]
```

Comment:
Do these have a shared base class that is directly importable? Seems like this requires a bit of import gymnastics not sure worth it

Reply:
they share abc.Iterable ...

pandas\io\sas\sasreader.py:93: error: "Iterator[Any]" has no attribute "read"
pandas\io\sas\sasreader.py:94: error: "Iterator[Any]" has no attribute "close"

many options:

Use protocol (python 3.8) from typing_extensions
define abc base class with read and close methods (add comment to use Protocol from py3.8)
use dynamically typed code, i.e. reader: Any
use ignore
leave as is with import gymnastics and Union.

we normally try to avoid most of these.

Reply:
> define abc base class with read and close methods (add comment to use Protocol from py3.8)

gone with this option.